features/support/github.js

```diff
@@ -142,6 +150,107 @@ function createPullRequest(branch, repoOwner, repoName) {
     });
 }
 
+/**
+ * Creates a lightweight tag.
+ * @method createTag
+ * @param  {String}   tag               Name of the tag
+ * @param  {String}   branch            The branch to create the file in
+ * @param  {String}   [repoOwner]       Owner of the repository
+ * @param  {String}   [repoName]        Name of the repository
+ * @return {Promise}
+ */
+function createTag(tag, branch, repoOwner, repoName) {
+    const owner = repoOwner || 'screwdriver-cd-test';
+    const repo = repoName || 'functional-git';
+
+    return octokit.git.getRef({
+        owner,
+        repo,
+        ref: `heads/${branch}`
+    })
+        .then((referenceData) => {
+            const sha = referenceData.data.object.sha;
+
+            return octokit.git.createRef({
+                owner,
+                repo,
+                ref: `refs/tags/${tag}`,
+                sha
+            });
+        })
+        .catch((err) => {
+            Assert.strictEqual(err.status, 422);
+        });
+}
+
+/**
+ * Creates a annotated tag.
+ * @method createAnnotatedTag
+ * @param  {String}   tag               Name of the tag
+ * @param  {String}   branch            The branch to create the file in
+ * @param  {String}   [repoOwner]       Owner of the repository
+ * @param  {String}   [repoName]        Name of the repository
+ * @return {Promise}
+ */
+function createAnnotatedTag(tag, branch, repoOwner, repoName) {
+    const owner = repoOwner || 'screwdriver-cd-test';
+    const repo = repoName || 'functional-git';
+
+    return octokit.git.getRef({
+        owner,
+        repo,
+        ref: `heads/${branch}`
+    })
+        .then((referenceData) => {
+            const sha = referenceData.data.object.sha;
+
+            return octokit.git.createTag({
+                owner,
+                repo,
+                tag,
+                message: 'this is annotated tag',
+                object: sha,
+                type: 'commit',
+                tagger: {
+                    name: 'test',
+                    email: 'test@example.com',
+                    date: '2019-10-09T15:00:00+09:00'
+                }
+            })
+                .then(response => octokit.git.createRef({
```

Comment:
This `then()` should be outside of this function.

Reply:
Why do you think this `then()` should be outside of this function ?
I think this `then()` can be in this function because `createTag()` makes a tag object only and it is difficult to understand for us.

Reply:
My intention is like following:
```
return octokit.git.getRef({
    (snip)
}).then((referenceData) => {
    const sha = referenceData.data.object.sha;

    return octokit.git.createTag({
                (snip)
    })
}).then(response => octokit.git.createRef({
    ...
    }));
}).catch(() => {...});
```

Reply:
I had a misunderstanding. I fixed it.